web-api/tests/silo_management_api.rs

```diff
@@ -0,0 +1,135 @@
+use xayn_web_api::Ingestion;
+use xayn_web_api_db_ctrl::{OperationResult, Tenant};
+use xayn_web_api_shared::request::TenantId;
+
+#[derive(Deserialize)]
+struct ManagementResponse {
+    results: Vec<OperationResult>,
+}
+
+#[tokio::test]
+async fn test_tenants_can_be_created() {
+    test_app::<Ingestion, _>(
+        Some(toml! {
+            [tenants]
+            enable_legacy_tenant = false
+        }),
+        |client, url, services| async move {
+            let test_id = services.test_id.as_str();
+            let make_id = |suffix| format!("{test_id}_{suffix}").parse::<TenantId>();
+            let ManagementResponse { results } = send_assert_json(
+                &client,
+                client.post(url.join("_silo_management_api")?).json(&json!({
+                    "operations": [
+                        { "CreateTenant": { "tenant_id": make_id("1")? }},
+                        { "CreateTenant": { "tenant_id": make_id("3")?, "is_legacy_tenant": true }},
+                        { "CreateTenant": { "tenant_id": make_id("1")? }},
+                        { "ListTenants": {} },
+                        { "DeleteTenant": { "tenant_id": make_id("3")? }},
+                        { "ListTenants": {} },
+                        { "DeleteTenant": { "tenant_id": make_id("3")? }},
+                    ]
+                })).build()?,
+                StatusCode::OK,
+            )
+            .await;
+
+            let mut results = results.into_iter();
+
+            assert_eq!(
+                results.next().unwrap(),
+                OperationResult::CreateTenant {
+                    tenant: Tenant {
+                        tenant_id: make_id("1")?,
+                        is_legacy_tenant: false,
+                    }
+                }
+            );
+            assert_eq!(
+                results.next().unwrap(),
+                OperationResult::CreateTenant {
+                    tenant: Tenant {
+                        tenant_id: make_id("3")?,
+                        is_legacy_tenant: true,
+                    }
+                }
+            );
+            assert!(matches!(
+                &results.next().unwrap(),
+                OperationResult::Error { .. }
+            ));
+            let OperationResult::ListTenants { tenants } = results.next().unwrap() else {
+                panic!("failed ot list tenants");
+            };
+            assert_eq!(
+                tenants.iter().collect::<HashSet<_>>(),
+                [
+                    services.tenant.clone(),
+                    Tenant {
+                        tenant_id: make_id("1")?,
+                        is_legacy_tenant: false,
+                    },
+                    Tenant {
+                        tenant_id: make_id("3")?,
+                        is_legacy_tenant: true,
+                    }
+                ]
+                .iter()
+                .collect::<HashSet<_>>()
```

Comment:
:question: shouldn't type inference work here? same question for other collects in the test.
```suggestion
                .into(),
```